datalad_next/create_sibling_webdav.py

```diff
@@ -0,0 +1,437 @@
+            constraints=EnsureStr() | EnsureNone()),
+        credential=Parameter(
+            args=("--credential",),
+            doc="""name of the credential providing a user/password credential
+            to be used for authorization. The credential can be supplied via
+            configuration setting 'datalad.credential.<name>.user|password', or
+            environment variable DATALAD_CREDENTIAL_<NAME>_USER|PASSWORD, or will
+            be queried from the active credential store using the provided
+            name. If none is provided, the last-used token for the
+            API URL realm will be used.""",
+        ),
+        existing=Parameter(
+            args=("--existing",),
+            constraints=EnsureChoice('skip', 'error', 'reconfigure', None),
+            metavar='MODE',
+            doc="""action to perform, if a (storage) sibling is already
+            configured under the given name and/or a target already exists.
+            In this case, a dataset can be skipped ('skip'), an existing target
+            repository be forcefully re-initialized, and the sibling
+            (re-)configured ('reconfigure'), or the command be instructed to
+            fail ('error').""", ),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+        storage_sibling=Parameter(
+            args=("--storage-sibling",),
+            dest='storage_sibling',
+            metavar='MODE',
+            constraints=EnsureChoice(
+                'yes', 'export', 'only', 'only-export', 'no'),
+            doc="""Both Git history and file content can be hosted on WEBDAV.
+            With 'yes', a storage sibling and a Git repository
+            sibling are created ('yes').
+            Alternatively, creation of the storage sibling can be disabled
+            ('no'),
+            or a storage sibling can be created only and no Git sibling
+            ('only').
+            The storage sibling can be set up as a standard git-annex special
+            remote that is capable of storage any number of file versions,
+            using a content hash based file tree ('yes'|'only'), or
+            as an export-type special remote, that can only store a single
+            file version corresponding to one unique state of the dataset,
+            but using a human-readable data data organization on the WEBDAV
+            remote that matches the file tree of the dataset
+            ('export'|'only-export')."""),
+    )
+
+    @staticmethod
+    @datasetmethod(name='create_sibling_webdav')
+    @eval_results
+    def __call__(
+            url: str,
+            *,
+            dataset: Optional[Union[str, Dataset]] = None,
+            name: Optional[str] = None,
+            storage_name: Optional[str] = None,
+            storage_sibling: Optional[str] = 'yes',
+            credential: Optional[str] = None,
+            existing: Optional[str] = None,
+            recursive: Optional[bool] = False,
```

Comment:
`Optional` should be removed. It is not correctly reflecting the type of `recursive`, since `recursive` is always a `bool`.

```suggestion
            recursive: bool = False,
```